Eslint Development Issues

Vliegenier04 opened the pull request:
@jejebecarte @Amxgh Please have a look at these ESLint error's that I have solved. Thanks in advance.

jejebecarte commented:
Hello! Looks mostly good! I took a quick look and a some things popped out:
- you should use consola (this doesn't have winston yet) for those log statements you used `eslint-disable` for - it's best to keep it consistent and also avoid `@ignore`s whenever possible
- looking at the `package.json` for the above I saw you ended up using `g`... please remove is possible :pray:
- you might want to check your use of async/await in some area - `handleStatsCommand` for example. You use it asynchronously but based on it's implementation it doesn't look like it's actually necessary. There might be other bits like that too.
- you'll probably want to move `joindata.json` out of the root folder
- be cautious of synchronous file reading - I think I did that back when I wrote the blacklist logic but it should really be async

The file structure is looking good though! Big fan.

P.S. - I would also recommend squashing when you merge so the timeline is neater.

Vliegenier04 commented:
@jejebecarte 
- I have removed ``g`` from ``package.json``. 
- I have removed the ``console.log`` statements to include ``logger.log`` with their respectable imports on the right place. 
- ``joindata.json`` is still in root folder since I've looked into it and figured it is somewhere the same as ``.env`` file and thus is better at it's place in ``root``.
- Synchronous file reading does not really impact this function as I see. 
- ``Async`` & ``Await`` for ``handleStatsCommand`` is required since we are working in a Event file. It breaks if it isn't Asynchronous. 

I don't know what squashing is ;)

jejebecarte commented:
Ok awesome thanks.

> joindata.json is still in root folder since I've looked into it and figured it is somewhere the same as .env file and thus is better at it's place in root.

I don't quite understand what you mean by this...

> Synchronous file reading does not really impact this function as I see.

Pretty sure it does. If I find it I'll point it out in the GH code comment thing.

See more info on squashing branches [here](https://www.git-tower.com/learn/git/faq/git-squash).

jejebecarte commented on src/events/mineflayer/chat/member-join-leave.ts:
```diff
@@ -54,57 +53,52 @@ async function processJoinEvent(bot: any, playerName: string, mojangProfile: any
         fs.writeFileSync(filePath, '{}');
     }
-    const joinData = require(filePath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
```
This is synchronous

Vliegenier04 replied:
Fixed.

jejebecarte commented on src/events/mineflayer/chat/member-join-leave.ts:
```diff
@@ -153,24 +146,19 @@ async function processLeaveEvent(bot: any, playerName: string, mojangProfile: an
 
     const filePath = path.resolve(__dirname, 'joindata.json');
-    const oldJoinData = require(filePath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
```
This is synchronous

Vliegenier04 replied:
Fixed.

jejebecarte commented on src/events/mineflayer/chat/member-kick.ts:
```diff
@@ -36,70 +40,46 @@ export default {
+                }
+
+                const filePath = path.resolve(__dirname, 'joindata.json');
+                const fileContent = fs.readFileSync(filePath, 'utf8');
```
This is synchronous

jejebecarte commented on src/classes/bot.ts:
```diff
@@ -173,11 +173,11 @@ class Bot {
         const rest = new REST({ version: '10' }).setToken(env.DISCORD_TOKEN);
 
         try {
```
I'm pretty sure this is unnecessary but I may be wrong - commands should already be updated on restart [here](https://github.com/MiscGuild/bridge/blob/398ea8ba738866fcf7c727266e309870b8a8965e/src/discord/events/ready.ts#L8)

Vliegenier04 replied:
This is a feature, since Discord api doesn't refresh after reboot.

Vliegenier04 replied:
Removed.
